docs/resources/aws_ec2_transit_gateway_route_table_association.md

```diff
@@ -0,0 +1,81 @@
+
+## Syntax
+
+Ensure that a transit gateway route table ID exists.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should exist }
+    end
+
+## Parameters
+
+`transit_gateway_route_table_id` _(required)_
+
+For additional information, see the [AWS documentation on transit gateway route table association](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewayroutetableassociation.html).
+
+## Properties
+
+| Property | Description|
+| --- | --- |
+| transit_gateway_route_table_id | The ID of the transit gateway route table. |
+| resource_type | The resource type. Valid values are: `vpc`, `vpn`, `direct-connect-gateway`, `peering`, and `connect`. |
+| resource_id | The ID of the resource. |
+| state | The possible states of the route table are: `available`, `deleting`, `deleted`, and `pending`. |
+
+## Examples
+
+### Ensure a transit gateway route table ID is available.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      its('transit_gateway_attachment_id') { should eq ''TRANSIT_GATEWAY_ATTACHMENT_ID'' }
+    end
+
+### Ensure that the state is available or deleted.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+        its('state') { should eq 'available' }
+    end
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+The controls will pass if the `describe` method returns at least one result.
+
+### exist
+
+Use `should` to test that the entity exists.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should exist }
+    end
+
+Use `should_not` to test an entity that should not exist.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should_not exist }
+    end
+
+### be_available
+
+Use `should` to check if the transit gateway route table ID is available.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should be_available }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `ec2:client:get_transit_gateway_route_table_associations` action with `Effect` set to `Allow`.
+
+See the [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html) documentation for additional information.
```

Comment:
<!--StartFragment-->For additional information, check out the [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html) documentation.<br class="Apple-interchange-newline"><!--EndFragment-->